src/Akeneo/Pim/Enrichment/Component/Product/Commands/GroupProductsHandler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+
+namespace Akeneo\Pim\Enrichment\Component\Product\Commands;
+
+use Akeneo\Pim\Enrichment\Component\Product\Query\FindProductIdentifiersInterface;
+use Akeneo\Pim\Enrichment\Component\Product\Repository\GroupRepositoryInterface;
+use Akeneo\Pim\Enrichment\Component\Product\Repository\ProductRepositoryInterface;
+use Akeneo\Tool\Component\StorageUtils\Saver\BulkSaverInterface;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class GroupProductsHandler
+{
+    private FindProductIdentifiersInterface $findGroupProductIdentifiers;
+    private GroupRepositoryInterface $groupRepository;
+    private BulkSaverInterface $productSaver;
+    private ProductRepositoryInterface $productRepository;
+
+
+    public function __construct(FindProductIdentifiersInterface $getGroupProductIdentifiers, GroupRepositoryInterface $groupRepository, BulkSaverInterface $productSaver, ProductRepositoryInterface $productRepository)
+    {
+        $this->findGroupProductIdentifiers = $getGroupProductIdentifiers;
+        $this->groupRepository = $groupRepository;
+        $this->productSaver = $productSaver;
+        $this->productRepository = $productRepository;
+    }
+
+    public function handle(GroupProductsCommand $updateProductsToGroupCommand)
+    {
+        $currentProductIds = $updateProductsToGroupCommand->productIds();
```

Comment:
```suggestion
        $newProductIds = $updateProductsToGroupCommand->productIds();
```

current is confusing. For me current refers that there is actually in the DB. So what you call old.

I see you already have a $newProductIds below. I propose to change this one by $addedProductIds